src/azul/indexer.py

```diff
@@ -176,8 +196,60 @@ def merge(cur_bundles, new_bundles):
             except KeyError:
                 bundle = new_bundle
             else:
-                bundle = new_bundle if new_bundle['version'] >= cur_bundle['version'] else cur_bundle
+                bundle = new_bundle if ((new_bundle['version'] >= cur_bundle['version']) or
+                                        new_bundle['admin_deleted']) else cur_bundle
             assert bundles.setdefault(bundle_uuid, bundle) is bundle
         for bundle in cur_bundles_by_id.values():
             assert bundles.setdefault(bundle['uuid'], bundle) is bundle
         return list(bundles.values())
+
+    def delete(self, dss_notification: Mapping[str, Any]) -> None:
+        bundle_uuid = dss_notification['match']['bundle_uuid']
+        bundle_version = dss_notification['match']['bundle_version']
+
+        es_client = self.properties.elastic_search_client
+
+        search_query = {
+            "query": {
+                "bool": {
+                    "must": [
+                        {
+                            "term": {
```

Comment:
I think this query could be simplified using one `terms` branch instead of two `term` branches.